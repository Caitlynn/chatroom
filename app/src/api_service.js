import axios from 'axios';

const chatroomApi = 'http://localhost:8000/chatroom';

export const getMessages = async () => {
  const response = await axios.get(chatroomApi)
  return response.data
}

export const createMessage = async (message) => {
  const response = await axios.post(chatroomApi, message)
  return response.data
}
