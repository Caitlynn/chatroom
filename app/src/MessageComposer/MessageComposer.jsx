import React, { Component } from 'react'
import PropTypes from 'prop-types'
import RaisedButton from 'material-ui/RaisedButton'
import TextField from 'material-ui/TextField'
import { createMessage } from '../api_service'
import './MessageComposer.css'

class MessageComposer extends Component {
  constructor(props){
    super(props)
    this.state = {
      message: '',
    }
  }

  onDraftChange = (event) => {
    this.setState({
      message: event.target.value
    })
  }

  postMessage = async (event) => {
    event.preventDefault()
    await createMessage({
      message: this.state.message,
      username: this.props.username
    })
    this.setState({ message: '' })
  }

  render() {
    const { message } = this.state
    return (
      <div className="Message-composer">
        <form className="Message-composer_form" onSubmit={this.postMessage}>
          <TextField
            className="Message-composer_chat-input"
            value={message}
            onChange={this.onDraftChange}
            placeholder="Enter message..."
          />
          <RaisedButton
            className="Message-composer_sendbutton"
            label="Send"
            type="submit"
          />
        </form>
      </div>
    )
  }
}

MessageComposer.propTypes = {
  username: PropTypes.string.isRequired
}

export default MessageComposer