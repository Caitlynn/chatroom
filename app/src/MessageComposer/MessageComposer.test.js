import React from 'react';
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import * as apiService from '../api_service'
import MessageComposer from './MessageComposer'

//to solve enzyme internal error
configure({ adapter: new Adapter() })

describe('Message Composer', () => {
  it('should be defined', () => {
    expect(MessageComposer).toBeDefined()
  })

  it('should render correctly', () => {
    const messageComposer = shallow(<MessageComposer username='test user'/>)
    expect(messageComposer).toMatchSnapshot()
  })

  it('should call api when form is submitted', () => {
    apiService.createMessage = jest.fn()
    const messageComposer = shallow(<MessageComposer username='test user' />)
    const mockEvent = { preventDefault: jest.fn() }
    messageComposer.find('.Message-composer_form').simulate('submit', mockEvent)
    expect(mockEvent.preventDefault.mock.calls.length).toBe(1)
    expect(apiService.createMessage.mock.calls.length).toBe(1)
  })
})