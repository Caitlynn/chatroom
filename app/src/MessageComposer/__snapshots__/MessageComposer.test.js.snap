// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Composer should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MessageComposer
    username="test user"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="Message-composer_form"
        onSubmit={[Function]}
>
        <TextField
                className="Message-composer_chat-input"
                disabled={false}
                floatingLabelFixed={false}
                fullWidth={false}
                multiLine={false}
                onChange={[Function]}
                placeholder="Enter message..."
                rows={1}
                type="text"
                underlineShow={true}
                value=""
        />
        <RaisedButton
                className="Message-composer_sendbutton"
                disabled={false}
                fullWidth={false}
                label="Send"
                labelPosition="after"
                primary={false}
                secondary={false}
                type="submit"
        />
</form>,
      "className": "Message-composer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <TextField
            className="Message-composer_chat-input"
            disabled={false}
            floatingLabelFixed={false}
            fullWidth={false}
            multiLine={false}
            onChange={[Function]}
            placeholder="Enter message..."
            rows={1}
            type="text"
            underlineShow={true}
            value=""
/>,
          <RaisedButton
            className="Message-composer_sendbutton"
            disabled={false}
            fullWidth={false}
            label="Send"
            labelPosition="after"
            primary={false}
            secondary={false}
            type="submit"
/>,
        ],
        "className": "Message-composer_form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "Message-composer_chat-input",
            "disabled": false,
            "floatingLabelFixed": false,
            "fullWidth": false,
            "multiLine": false,
            "onChange": [Function],
            "placeholder": "Enter message...",
            "rows": 1,
            "type": "text",
            "underlineShow": true,
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "Message-composer_sendbutton",
            "disabled": false,
            "fullWidth": false,
            "label": "Send",
            "labelPosition": "after",
            "primary": false,
            "secondary": false,
            "type": "submit",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="Message-composer_form"
          onSubmit={[Function]}
>
          <TextField
                    className="Message-composer_chat-input"
                    disabled={false}
                    floatingLabelFixed={false}
                    fullWidth={false}
                    multiLine={false}
                    onChange={[Function]}
                    placeholder="Enter message..."
                    rows={1}
                    type="text"
                    underlineShow={true}
                    value=""
          />
          <RaisedButton
                    className="Message-composer_sendbutton"
                    disabled={false}
                    fullWidth={false}
                    label="Send"
                    labelPosition="after"
                    primary={false}
                    secondary={false}
                    type="submit"
          />
</form>,
        "className": "Message-composer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <TextField
              className="Message-composer_chat-input"
              disabled={false}
              floatingLabelFixed={false}
              fullWidth={false}
              multiLine={false}
              onChange={[Function]}
              placeholder="Enter message..."
              rows={1}
              type="text"
              underlineShow={true}
              value=""
/>,
            <RaisedButton
              className="Message-composer_sendbutton"
              disabled={false}
              fullWidth={false}
              label="Send"
              labelPosition="after"
              primary={false}
              secondary={false}
              type="submit"
/>,
          ],
          "className": "Message-composer_form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "Message-composer_chat-input",
              "disabled": false,
              "floatingLabelFixed": false,
              "fullWidth": false,
              "multiLine": false,
              "onChange": [Function],
              "placeholder": "Enter message...",
              "rows": 1,
              "type": "text",
              "underlineShow": true,
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "Message-composer_sendbutton",
              "disabled": false,
              "fullWidth": false,
              "label": "Send",
              "labelPosition": "after",
              "primary": false,
              "secondary": false,
              "type": "submit",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
